scene/resources/particle_process_material.cpp

```diff
@@ -750,57 +759,55 @@ void ParticleProcessMaterial::_update_shader() {
 		code += "	vec3 direction_yz = vec3(0.0, sin(angle2_rad), cos(angle2_rad));\n";
-		code += "	direction_yz.z = direction_yz.z / max(0.0001,sqrt(abs(direction_yz.z))); // better uniform distribution\n";
+		code += "	direction_yz.z = direction_yz.z / max(0.0001, sqrt(abs(direction_yz.z))); // Better uniform distribution.\n";
 		code += "	vec3 spread_direction = vec3(direction_xz.x * direction_yz.z, direction_yz.y, direction_xz.z * direction_yz.z);\n";
 		code += "	vec3 direction_nrm = length(direction) > 0.0 ? normalize(direction) : vec3(0.0, 0.0, 1.0);\n";
-		code += "	// rotate spread to direction\n";
+		code += "	// Rotate spread to direction.\n";
 		code += "	vec3 binormal = cross(vec3(0.0, 1.0, 0.0), direction_nrm);\n";
 		code += "	if (length(binormal) < 0.0001) {\n";
-		code += "		// direction is parallel to Y. Choose Z as the binormal.\n";
+		code += "		// Direction is parallel to Y. Choose Z as the binormal.\n";
 		code += "		binormal = vec3(0.0, 0.0, 1.0);\n";
 		code += "	}\n";
 		code += "	binormal = normalize(binormal);\n";
 		code += "	vec3 normal = cross(binormal, direction_nrm);\n";
 		code += "	spread_direction = binormal * spread_direction.x + normal * spread_direction.y + direction_nrm * spread_direction.z;\n";
 		code += "	return normalize(spread_direction);\n";
 	}
-	code += "}\n";
+	code += "}\n\n";
 
-	code += "vec3 process_radial_displacement(DynamicsParameters param, float lifetime, inout uint alt_seed, mat4 transform, mat4 emission_transform, float delta){\n";
+	code += "vec3 process_radial_displacement(DynamicsParameters param, float lifetime, inout uint alt_seed, mat4 transform, mat4 emission_transform, float delta) {\n";
 	code += "	vec3 radial_displacement = vec3(0.0);\n";
-	code += "	if (delta < 0.001){\n";
+	code += "	if (delta < 0.001) {\n";
 	code += "		return radial_displacement;\n";
 	code += "	}\n";
 	code += "	float radial_displacement_multiplier = 1.0;\n";
 	if (tex_parameters[PARAM_RADIAL_VELOCITY].is_valid()) {
 		code += "	radial_displacement_multiplier = texture(radial_velocity_curve, vec2(lifetime)).r;\n";
 	}
 	code += "	vec3 global_pivot = (emission_transform * vec4(velocity_pivot, 1.0)).xyz;\n";
-	code += "	if(length(transform[3].xyz - global_pivot) > 0.01){\n";
+	code += "	if (length(transform[3].xyz - global_pivot) > 0.01) {\n";
 	code += "		radial_displacement = normalize(transform[3].xyz - global_pivot) * radial_displacement_multiplier * param.radial_velocity;\n";
-	code += "	}else{radial_displacement = get_random_direction_from_spread(alt_seed, 360.0)* param.radial_velocity;} \n";
-	code += "	if (radial_displacement_multiplier * param.radial_velocity < 0.0){\n // Prevent inwards velocity to flicker once the point is reached.";
-	code += "		if (length(radial_displacement) > 0.01){\n";
-	code += "		radial_displacement = normalize(radial_displacement) * min(abs((radial_displacement_multiplier * param.radial_velocity)), length(transform[3].xyz - global_pivot) / delta);\n";
```

Comment:
So there was a bug here: The comment at the end of line 782 didn't end with `\n`, so line 783 was commented out. So I removed it, ensuring that the current logic is preserved.

But this might be a sign that the current logic is broken and needs review. CC @QbieShay